### PR TITLE
HPCC-25073 Support for code signing keys in vault

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/values.yaml [hunk 1]
```diff
@@ -197,9 +197,21 @@ vaults:
...
   esp:
 
+  ## The keys for code signing may be imported from the vault.  Multiple keys may be imported.
+  ## gpg keys may be imported as follows:
+  ## vault kv put secret/codeSign/gpg-private-key-1 passphrase=<passphrase> private=@<private_key_file1>
+  ## vault kv put secret/codeSign/gpg-private-key-2 passphrase=<passphrase> private=@<private_key_file2>
   codeSign:
-
+    - name: codesign-private-keys
+      url: http://${env.VAULT_SERVICE_HOST}:${env.VAULT_SERVICE_PORT}/secret/data/codeSign/${secret}
```
Comment (ghalliday) [inline helm/hpcc/values.yaml:206]: is this an example?  Should it be commoned out it the final PR?

**Comment by ghalliday** [comment]
@shamser I opened a new PR to improve the consistency of the categories: https://track.hpccsystems.com/browse/HPCC-25274.  It should probably be a separate PR, but it would be good for you to look at it.

File: helm/hpcc/values.yaml (hunk 1 above)
Comment (shamser) [reply helm/hpcc/values.yaml]: This URL works for the default vault configuration.  Users should be able to pull and start default the Hashicorp vault image, enable for k8,  upload the keys to the vault and use the service for signing and verification.  (There is a small caveat in that the egress rules will need to be updated).  Is there any harm in keeping this in?

**Comment by shamser** [comment]
> @shamser I opened a new PR to improve the consistency of the categories: https://track.hpccsystems.com/browse/HPCC-25274. It should probably be a separate PR, but it would be good for you to look at it.

HPCC-25274: It would make sense to rationalize secrets and vaults.  I'll start thinking about this.

The original PR was missing the small esp.yaml change.  Please can you review again.

File: helm/hpcc/values.yaml (hunk 1 above)
Comment (ghalliday) [reply helm/hpcc/values.yaml]: What will happen if vault is not installed?  Will the base coe stall waiting for a response, or generate an error if it is not available?

Comment (ghalliday) [reply helm/hpcc/values.yaml]: @shamser ?

Comment (shamser) [reply helm/hpcc/values.yaml]: If vault isn't installed or configured, an error will be written to the ESP log: Vault auth error - vault: codesign-private-keys - vault access token not provided.  I have commented out the vault configuration. @ghalliday

**Review by ghalliday — DISMISSED** [review]

**Review by ghalliday — APPROVED** [review]

**Pull request merged**